aptos-move/framework/aptos-token/sources/token.move

```diff
@@ -1614,8 +1618,18 @@ module aptos_token::token {
         assert!(table::contains(tokens, token_id), error::not_found(ENO_TOKEN_IN_TOKEN_STORE));
 
         let value = &mut table::borrow_mut(tokens, token_id).token_properties;
-
-        property_map::update_property_map(value, keys, values, types);
+        let i = 0;
+        while (i < vector::length(&keys)){
+            let key = vector::borrow(&keys, i);
+            assert_not_standard_reserved_property(key);
+            let new_pv = property_map::create_property_value_raw(*vector::borrow(&values, i), *vector::borrow(&types, i));
+            if (property_map::contains_key(value, key)) {
+                property_map::update_property_value(value, key, new_pv);
+            } else {
+                property_map::add(value, *key, new_pv);
+            };
+            i = i + 1;
+        };
```

Comment:
Instead of doing this, why not make `assert_non_...` take `&vector<string>`? before the loop?

Reply:
refactored